community/server/src/main/java/org/neo4j/server/rest/web/CypherService.java

```diff
@@ -118,6 +130,11 @@ public Response cypher(String body,
                 includePlan = result.getQueryExecutionType().requestedExecutionPlanDescription();
             }
 
+            if ( periodicCommitQuery )
+            {
+                handler.setTransaction( database.beginTx() );
```

Comment:
On the other hand, this transaction is used only for streaming back results and all the transactions opened for the `CypherService` have `AccessMode.Static.FULL`. So I don't think it is super important to fix it now, but we definitely improve it when we have security in place.